### ISSUE TITLE
Render HTML client-side

### ISSUE THREAD

**Issue by timhutton** [timhutton/twitter-archive-parser#65]
Currently we have a script that converts md files to html.

It is also possible to render the md on the client-side, using e.g.:

```
<!doctype html>
<html>
<head>
  <meta charset="utf-8"/>
  <title>Marked in the browser</title>
</head>
<body>
  <div id="content"></div>
  <script src="https://cdn.jsdelivr.net/npm/marked/marked.min.js"></script>
  <script>
    var md = "# Important #\nThis is some *markdown* text.";
    document.getElementById('content').innerHTML = marked.parse(md);
  </script>
</body>
</html>
```

We could eliminate `convert_to_html.py` by simply wrapping the md like that.

A more sophisticated solution could retain the separate md files and render them on demand:

```
<!doctype html>
<html>
<head>
  <meta charset="utf-8"/>
  <title>Marked in the browser</title>
</head>
<body>
  <div id="content"></div>
  <script src="https://cdn.jsdelivr.net/npm/marked/marked.min.js"></script>
  <script>
    var client = new XMLHttpRequest();
    client.open('GET', '/2013-12-01-Tweet-Archive-2013-12.md');
    client.onreadystatechange = function() {
        document.getElementById('content').innerHTML = marked.parse(client.responseText);
    }
    client.send();
  </script>
</body>
</html>
```

This could even be used to replace Jekyll completely which would be great from my point of view because a) maintaining the installation locally is a pain, b) running github actions just to publish your blog is crazy.

@sweh - thoughts?

**Comment by timhutton** [comment]
I tried it. It's slow to open the html file, even with a smallish archive of 3000 tweets. That was probably predictable. :)

We could do something more sophisticated with rendering the md on demand, or at least by splitting the output into linked pages. But I'm not enjoying this solution. Will close this.

**Comment by akhepcat** [comment]
FWIW,  I took @timhutton 's  HTML+JS  and modified it a bit to allow opening any of the MD files inline, using a quick script I wrote.  

Basically indexes all the markdown files into options, builds a webpage, and pops open either the node http server, or the python3 http server to serve that file.

Speed locally is pretty quick, and having the dropdown make it simple to switch between files.

https://github.com/akhepcat/Miscellaneous/blob/master/MDviewer.sh

(i toyed with writing it completely in nodejs, but then i thought better of it)
